examples/outgoing_http_requests/src/main.py

```diff
@@ -0,0 +1,79 @@
+from kybra import Async, blob, CanisterResult, ic, Principal, query, update, manual
+from kybra.canisters.management import management_canister
+from kybra.canisters.management.http import HttpResponse, HttpTransformArgs
+
+
+@update
+def xkcd() -> Async[HttpResponse]:
+    max_response_bytes = 1_000
+
+    # TODO this is just a heuristic for cost, might change when the feature is officially released: https://forum.dfinity.org/t/enable-canisters-to-make-http-s-requests/9670/130
+    cycle_cost_base = 400_000_000
+    cycle_cost_per_byte = 300_000  # TODO not sure on this exact cost
+    cycle_cost_total = cycle_cost_base + cycle_cost_per_byte * max_response_bytes
+
+    http_result: CanisterResult[HttpResponse] = yield management_canister.http_request({
+        'url': 'https://xkcd.com/642/info.0.json',
+        'max_response_bytes': max_response_bytes,
+        'method': {'get': None},
+        'headers': [],
+        'body': None,
+        'transform': {
+            'function': ((ic.id()), 'xkcd_transform'),
+            'context': bytes()
+        },
+    }).with_cycles(cycle_cost_total)
+
+    if http_result.err is not None:
+        if http_result.err:
+            ic.trap(http_result.err)
+        ic.trap('http_result had an error')
+
+    return http_result.ok
+
+
+@update
+def xkcd_raw() -> Async[manual[HttpResponse]]:
+    max_response_bytes = 1_000
+
+    # TODO this is just a heuristic for cost, might change when the feature is officially released: https://forum.dfinity.org/t/enable-canisters-to-make-http-s-requests/9670/130
+    cycle_cost_base = 400_000_000
+    cycle_cost_per_byte = 300_000  # TODO not sure on this exact cost
+    cycle_cost_total = cycle_cost_base + cycle_cost_per_byte * max_response_bytes
+
+    candid = """
```

Comment:
Could you use an f-string in here instead of .format? And if you could do it inline at the ic.candid_encode call site.

ChatGPT explains how to do multi-line f-strings:


Yes, Python f-strings can be multiline. To create a multiline f-string, you can use the \ character at the end of each line to indicate that the string will continue on the next line. You can also use triple quotes (""") or triple single quotes (''') to create a multiline f-string.

Here is an example of creating a multiline f-string using the \ character:

Copy code
# Define a variable to use in the f-string
name = "John Doe"

# Create a multiline f-string using the `\` character
formatted_string = f"My name is {name}\
and I am learning Python"

# Print the formatted string
print(formatted_string)
In this example, the \ character is used at the end of each line to indicate that the f-string will continue on the next line. This allows the f-string to span multiple lines, and the resulting string is printed to the console.

Here is an example of creating a multiline f-string using triple quotes:

Copy code
# Define a variable to use in the f-string
name = "John Doe"

# Create a multiline f-string using triple quotes
formatted_string = f"""
My name is {name}.
I am learning Python.
"""

# Print the formatted string
print(formatted_string)
In this example, triple quotes are used to create a multiline f-string. The resulting string is formatted with the value of the name variable at runtime, and the resulting formatted string is printed to the console.

You can also use triple single quotes to create a multiline f-string in Python. Here is an example of doing this:

Copy code
# Define a variable to use in the f-string
name = "John Doe"

# Create a multiline f-string using triple single quotes
formatted_string = f'''
My name is {name}.
I am learning Python.
'''

# Print the formatted string
print(formatted_string)
In this example, triple single quotes are used instead of triple quotes to create a multiline f-string. The resulting string is formatted with the value of the name variable at runtime, and the resulting formatted string is printed to the console.

Overall, Python f-strings can be easily made multiline by using the \ character, triple quotes, or triple single quotes. This allows you to create f-strings that span multiple lines, and can make your code easier to read and maintain.

Reply:
Best to `Create a multiline f-string using triple single quotes`